File: leetcode/python/55.跳跃游戏.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        ret = [False for _ in range(n)]
        ret[0] = True

        for i in range(n):
            if ret[i] == False:
                break
            if nums[i] + i + 1 >= n:
                return True
            for j in range(i + 1, i + 1 + nums[i]):
                if j + 1 + nums[j] >= n:
                    return True
                ret[j] = True
        return False

File: leetcode/golang/55.跳跃游戏.go
func canJump(nums []int) bool {
    n := len(nums);
    ret := make([]bool, n);
    ret[0] = true;

    i := 0;
    for {
        if i >= n || ret[i] == false {
            break
        }
        if nums[i] + i + 1 >= n {
            return true;
        }
        j := i + 1;
        for {
            if j >= nums[i] + i + 1 {
                break;
            }
            if nums[j] + j + 1 >= n {
                return true;
            }
            ret[j] = true;
            j += 1
        }
        i += 1;
    }
    return false;
}